pandas/core/frame.py

```diff
@@ -174,6 +174,18 @@
 
     .. versionadded:: 0.17.0
 
+validate : {None, '1:1', '1:m', 'm:1', 'm:m', "one_to_one", "one_to_many",
+            "many_to_one", "many_to_many"}, default None
```

Comment:
The formatting here (multiple lines for the type specification, to render nice in the html docs it has to fit on one line) currently does not work nice with numpydoc (see https://github.com/numpy/numpydoc/issues/87)

Putting just `string, default None` and then the options are in the list below is an alternative